lakefile.lean

```diff
@@ -120,8 +120,25 @@ def getSrcUri (mod : Module) : IO String := do
   | .some "file" => getFileUri mod
   | .some _ => throw <| IO.userError "$DOCGEN_SRC should be github, file, or vscode."
 
+target bibPrepass : FilePath := do
+  let exeJob ← «doc-gen4».fetch
+  let basePath := (←getWorkspace).root.buildDir / "doc"
+  let inputFile := (←getWorkspace).root.srcDir / "docs" / "references.bib"
+  let outputFile := basePath / "declarations" / "references.json"
+  exeJob.bindSync fun exeFile exeTrace => do
+    let inputTrace ← computeTrace inputFile <|> pure .nil
```

Comment:
Is combining with nil here needed? Furthermore how does lake react to being told to compute the trace of a file that does not exist in the error case?

Reply:
There is no documents for [Lake.computeTrace](https://leanprover-community.github.io/mathlib4_docs/Lake/Build/Trace.html#Lake.computeTrace). I've tried that if there is no `<|> pure .nil` then there will be errors:
```text
error: no such file or directory (error code: 2)
  file: .\.\.\docs\references.bib
```

Reply:
@tydeu how is supposed one deal with build steps that are optional depending on the existence of a file?

Reply:
@hargoniX Using `<|>` seems like a reasonable way of doing it. It might be wise to mix the computed trace with `nil` to decrease the possibility of a clash. Something like:

```
let inputTrace ← (mixTrace nilTrace <$> computeTrace inputFile) <|> pure .nil
```

I should probability add a convivence utility for doing this to the Lake library.